shardingsphere-operator/api/v1alpha1/shardingsphere_chaos_types.go

```diff
@@ -36,12 +38,202 @@ type ShardingSphereChaos struct {
+}
+
+// pod chaos
+
+type PodChaosAction string
+
+var (
+	PodFailureAction    PodChaosAction = "podFailure"
+	ContainerKillAction PodChaosAction = "ContainerKill"
+)
+
+type PodChaosSpec struct {
+	PodSelector `json:",inline"`
+	Action      PodChaosAction `json:"action"`
+	// +optional
+	PodActionParam *PodActionParam `json:",inline"`
+}
+
+type PodActionParam struct {
+	//+optional
+	PodFailure *PodFailureActionParams `json:"podFailure,omitempty"`
+	//+optional
+	ContainerKill *ContainerKillActionParams `json:"containerKill,omitempty"`
+}
+
+type PodFailureActionParams struct {
+	// +optional
+	Duration string `json:"duration,omitempty"`
+}
+
+type ContainerKillActionParams struct {
+	// +optional
+	ContainerNames []string `json:"containerNames,omitempty"`
+}
+
+//network chaos
+
+type NetworkChaosSpec struct {
+	Source PodSelector `json:",inline"`
+
+	// +optional
+	Duration *string `json:"duration,omitempty"`
+
+	//+optional
+	Direction Direction `json:"direction,omitempty"`
+
+	// +optional
+	Target *PodSelector `json:"target,omitempty"`
+
+	Action NetworkChaosAction `json:"action"`
+
+	// +optional
+	NetWorkParams *NetWorkParams `json:",inline"`
+}
+
+type NetWorkParams struct {
+	// +optional
+	Delay *DelayActionParams `json:"delay,omitempty"`
+	// +optional
+	Loss *LossActionParams `json:"loss,omitempty"`
+	// +optional
+	Duplicate *DuplicateActionParams `json:"duplicate,omitempty"`
+	// +optional
+	Corrupt *CorruptActionParams `json:"corrupt,omitempty"`
+}
+
+type DelayActionParams struct {
+	// +optional
+	Latency string `json:"latency,omitempty"`
+	// +optional
+	Correlation string `json:"correlation,omitempty"`
+	// +optional
+	Jitter string `json:"jitter,omitempty"`
+}
+
+type LossActionParams struct {
+	// +optional
+	Loss string `json:"loss,omitempty"`
+	// +optional
+	Correlation string `json:"correlation,omitempty"`
+}
+
+type DuplicateActionParams struct {
+	// +optional
+	Duplicate string `json:"duplicate,omitempty"`
+	// +optional
+	Correlation string `json:"correlation,omitempty"`
+}
+
+type CorruptActionParams struct {
+	// +optional
+	Corrupt string `json:"corrupt,omitempty"`
+	// +optional
+	Correlation string `json:"correlation,omitempty"`
+}
+
+type NetworkChaosAction string
+
+const (
+	DelayAction NetworkChaosAction = "delay"
+
+	LossAction NetworkChaosAction = "loss"
+
+	DuplicateAction NetworkChaosAction = "duplicate"
+
+	CorruptAction NetworkChaosAction = "corrupt"
+
+	PartitionAction NetworkChaosAction = "partition"
+)
+
+type Direction string
+
+const (
+	To Direction = "to"
+
+	From Direction = "from"
+
+	Both Direction = "both"
+)
+
+//selector
+
+type PodSelector struct {
+	// +optional
+	Namespaces []string `json:"namespaces,omitempty"`
+
+	// +optional
+	LabelSelectors map[string]string `json:"labelSelectors,omitempty"`
+
+	// +optional
+	AnnotationSelectors map[string]string `json:"annotationSelectors,omitempty"`
+
+	// +optional
+	Nodes []string `json:"nodes,omitempty"`
```

Comment:
What's the difference between `Nodes` and `NodeSelectors` ?

Reply:
`NodesSelector`:  Specifies the Labels of the Node to which the experiment target Pod belongs.
`Node`: Specifies the experiment nodes